src/main/common/sensor_alignment.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "axis.h"
+#include "maths.h"
+
+typedef enum {
+    ALIGN_DEFAULT = 0, // driver-provided alignment
+
+    CW0_DEG = 1,
+    CW90_DEG = 2,
+    CW180_DEG = 3,
+    CW270_DEG = 4,
+    CW0_DEG_FLIP = 5,
+    CW90_DEG_FLIP = 6,
+    CW180_DEG_FLIP = 7,
+    CW270_DEG_FLIP = 8,
+
+    ALIGN_CUSTOM = 9, // arbitrary sensor angles, e.g. for external sensors
+} sensor_align_e;
+
+typedef union sensorAlignment_u {
+    // value order is the same as axis_e
+
+    int16_t raw[XYZ_AXIS_COUNT];
+    struct {
+        int16_t roll;
+        int16_t pitch;
+        int16_t yaw;
+    } values;
+} sensorAlignment_t;
+
+
+#define CUSTOM_ALIGN_CW0_DEG         ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 0,   .yaw = 0    }})
+#define CUSTOM_ALIGN_CW90_DEG        ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 0,   .yaw = 90   }})
+#define CUSTOM_ALIGN_CW180_DEG       ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 0,   .yaw = 180  }})
+#define CUSTOM_ALIGN_CW270_DEG       ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 0,   .yaw = 270  }})
+#define CUSTOM_ALIGN_CW0_DEG_FLIP    ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 180, .yaw = 0    }})
+#define CUSTOM_ALIGN_CW90_DEG_FLIP   ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 180, .yaw = 90   }})
+#define CUSTOM_ALIGN_CW180_DEG_FLIP  ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 180, .yaw = 180  }})
+#define CUSTOM_ALIGN_CW270_DEG_FLIP  ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 180, .yaw = 270  }})
+
+void buildRotationMatrixFromAlignment(const sensorAlignment_t* alignment, fp_rotationMatrix_t* rm);
+void buildAlignmentFromLegacyRotation(sensorAlignment_t* sensorAlignment, sensor_align_e rotation);
```

Comment:
Some of these functions seem to be only used in `sensor_alignment.c` and potentially tests. Can you please remove them from here, and make them `static` or `STATIC_UNIT_TESTED`. Besides a speed up in build time, having functions declared static makes code easier to understand, as it makes it obvious that they will not be called from outside of the compilation unit.

Reply:
yes, as noted some code from the old PR is still here as I've not tackled the MSP changes yet, see below..

Reply:
done